### DISCUSSION TITLE
Request: Example and Test Scenario for MultiplayerSpawner and MultiplayerSynchronizer

### DISCUSSION THREAD

**Discussion by Vitorsilveira31** [SakulFlee/Godot-WebRTC-Match-Maker#173] [General]
**Is your feature request related to a problem? Please describe.**
I'm seeking assistance in understanding and implementing the MultiplayerSpawner and MultiplayerSynchronizer modules within the Godot-WebRTC-Match-Maker library for my project.

**Describe the solution you'd like**
I would like to have a simple and clear example demonstrating the usage of the MultiplayerSpawner and MultiplayerSynchronizer in the RPC demo project.

**Describe alternatives you've considered**
I haven't explored any alternatives yet, as I'm specifically interested in utilizing these modules within the Godot-WebRTC-Match-Maker library.

**Additional context**
I appreciate any help provided, as it will greatly aid me in incorporating these features into my project. Thank you!

**Comment by SakulFlee** [comment]
Hi, thanks for reaching out! :)

There currently isn't an example utilizing Godot's Multiplayer-Nodes ([MultiplayerSynchronizer](https://docs.godotengine.org/en/stable/classes/class_multiplayersynchronizer.html) and [MultiplayerSpawner](https://docs.godotengine.org/en/stable/classes/class_multiplayerspawner.html)) as ... well ... it's Godot's nodes.
They utilize the [MultiplayerPeer](https://docs.godotengine.org/en/stable/classes/class_multiplayerpeer.html) under the hood. So by simply setting the `Multiplayer.MultiplayerPeer` property correctly, you _should_ be able to use those nodes as well!
You can see an example of setting this with the Match Maker in the [RPC Demo](https://github.com/SakulFlee/Godot-WebRTC-Match-Maker/tree/main/Godot%20Project/Demos/RPC).
Specifically, [RPC.cs:16](https://github.com/SakulFlee/Godot-WebRTC-Match-Maker/blob/b16ad7e0e38080fef58b760ae6c37543c078b585/Godot%20Project/Demos/RPC/RPC.cs#L16)

As a guide for Godot's nodes, you may refer to [this blog article from Godot](https://godotengine.org/article/multiplayer-in-godot-4-0-scene-replication/)!
It's in GDScript, but it's almost 1:1 with C# :)

I personally found Godot's network nodes very lacking though, which lead me to implement my own protocol for the [Game demo](https://github.com/SakulFlee/Godot-WebRTC-Match-Maker/tree/main/Godot%20Project/Demos/Game) as it's much better performing and is utilizing features of WebRTC (like channels) better.

Let me know if I can provide more help!
